src/main/groovy/com/github/jengelman/gradle/plugins/shadow/internal/DefaultR8Configuration.groovy

```diff
@@ -0,0 +1,16 @@
+package com.github.jengelman.gradle.plugins.shadow.internal
+
+class DefaultR8Configuration implements R8Configuration {
+
+    private final Collection<String> additionalRules = new ArrayList<>();
+
+    @Override
+    Collection<String> getRules() {
+        return additionalRules
+    }
+
+    @Override
+    void rule(String rule) {
```

Comment:
It would be nice to allow load rules from config file, maybe something like `rules(file("proguard.conf"))`

Reply:
Makes sense, will add this.